Dev sonia

soniakbendre opened the pull request:
implemented methods in simulation.py!

soniakbendre committed:
Update simulation.py

implemented invoke_event method

soniakbendre committed:
Update simulation.py

added imports for datetime for attributes start_time and stop_time

implemented start_simulation and stop_simulation

soniakbendre committed:
Update simulation.py

added implementations for is_simulation_running, get_simulation_time, get_simulation_step, and start_capture (from typhoon hil api)

soniakbendre committed:
Update simulation.py

added stop_signal attribute

implemented initialize, set_stop_signal, clear_stop_signal, and get_stop_signal

soniakbendre committed:
Update simulation.py

added docstrings, implemented capture_in_progress, unimplemented start_capture

soniakbendre committed:
Update classes.md

added return type to is_capture_in_progress

soniakbendre committed:
Update automator.py

soniakbendre committed:
Update automator.py

implemented set_data_logger_filename

soniakbendre committed:
Update automator.py

implemented add_digital_capture_signals

eggebee6 commented on src/typhoon_automator/automator.py:
```diff
@@ -151,7 +154,9 @@ def add_data_logger_signals(
         :raises ValueError: The loaded schematic does not contain one or more of the signal names
         """
-        raise NotImplementedError()
+        if not signals or len(signals) == 0:
```
This is correct, but we can tighten it up just a bit:
```suggestion
        if (not signals) or (len(signals) < 1):
```

soniakbendre committed:
Update automator.py

implemented add_analog_capture_signals and add_digital_capture_signals
(in the last commit it was add_data_logger_signals)

eggebee6 commented on src/typhoon_automator/automator.py:
```diff
@@ -151,7 +154,9 @@ def add_data_logger_signals(
-        raise NotImplementedError()
+        if not signals or len(signals) == 0:
+            raise ValueError('The loaded schematic does not contain one or more of the signal names')
+        self._data_log_signals = signals[:]
```
This method should add signals to the list, not replace the list:
```suggestion
        self._data_log_signals.append(signals)
```

eggebee6 commented on src/typhoon_automator/automator.py:
```diff
@@ -176,7 +181,10 @@ def add_analog_capture_signals(
+        if not signals or len(signals) == 0:
+            raise ValueError('The loaded schematic does not contain one or more of the signal names')
+        
+        self._analog_capture_signals = signals[:]
```
```suggestion
        self._analog_capture_signals.append(signals)
```

eggebee6 commented on src/typhoon_automator/automator.py:
```diff
@@ -186,8 +194,11 @@ def add_digital_capture_signals(
+        if not signals or len(signals) == 0:
+            raise ValueError('The loaded schematic does not contain one or more of the signal names')
 
+        self._digital_capture_signals = signals[:]
```
```suggestion
        self._digital_capture_signals.append(signals)
```

eggebee6 commented on src/typhoon_automator/simulation.py:
```diff
@@ -25,10 +31,17 @@ def __init__(
+        self.stop_signal = False
+        self.start_time: datetime = None
+        self.stop_time: datetime = None
+
```
I missed this in the last repo, so now's our chance to fix it:  Python doesn't have public/private class variables, so variables meant to be private are prefixed with an underscore (and public without an underscore).  `stop_signal`, `start_time`, and `stop_time` are all meant to be private and should have the underscore prefix.  If you can make this change (e.g. `self._stop_signal` and likewise), that'd be great!

soniakbendre committed:
Update simulation.py

deleted comment

eggebee6 commented on src/typhoon_automator/simulation.py:
```diff
@@ -39,22 +52,72 @@ def schedule_event(
+      
+        except AttributeError:
+          if not hasattr(event, "invoke"):
+            self._automator.log("Event object does not have an invoke method", level = logging.ERROR) # is this the right level?
```
`Error` is good!  It's a problem, so it's more severe than `Warning`, but the simulation _can_ continue, so it's not 'crash-and-burn' enough to be `Critical`.

Fun fact: this is related to the discussion we had about Python not having interfaces, meaning we have to write this code defensively.  Specifically, we use `event` expecting it to have an `invoke` method and a `message` string, and `AttributeError` gets thrown when it doesn't.

eggebee6 commented on src/typhoon_automator/simulation.py:
```diff
@@ -25,10 +31,17 @@ def __init__(
+        if scenario is None:
+            raise ValueError("Scenario cannot be None")
+
+        self.scenario = scenario
```
FYI:  this is alright for now, but we're going to have to re-visit it later.  There's a design change we didn't get to discuss which will affect this part of the code (briefly: Simulation no longer owns the scenario, Orchestrator owns the scenarios, Simulation just initializes them)

eggebee6 commented on src/typhoon_automator/simulation.py:
```diff
@@ -39,22 +52,72 @@ def schedule_event(
+          self._automator.log("Stopping simulation")
+          hil.stop_simulation()
+        else:
+          self_automator.log("Stop simulation called but simulation was not running", level = logger.WARNING)
```
Small typo
```suggestion
          self._automator.log("Stop simulation called but simulation was not running", level = logger.WARNING)
```

soniakbendre committed:
Update orchestrator.py

implemented add_scenario (i think)

eggebee6 reviewed (changes requested):
The main changes here would be appending to the signal lists rather than replacing them.  Everything else is small details.

soniakbendre committed:
Update automator.py

made changes to add_data_logger_signals, add_analog_capture_signals, and add_digital_capture_signals

soniakbendre committed:
Update simulation.py

added underscores to stop_signal, start_time, and stop_time attributes
fixed typos & added TODO

soniakbendre committed:
Update simulation.py

added some more docstrings

soniakbendre committed:
Update simulation.py

implemented schedule_event

soniakbendre committed:
Merge branch 'dev-sonia' of https://github.com/UWM-SEES/TyphoonAutomation-proto into dev-sonia
